tutorials/visual_field_plots.py

```
@@ -0,0 +1,178 @@
+                              layout: VisualFieldLayout | None = None,
+                              ax: plt.Axes | None = None,
+                              cmap: str = "coolwarm",
+                              levels: int = 200) -> plt.Axes:
+    """Create a smooth heat map from discrete visual field measurements."""
+    layout = layout or VisualFieldLayout.humphrey_24_2_with_fovea()
+    values = _ensure_array(values)
+
+    if values.size != layout.coordinates.shape[0]:
+        raise ValueError(
+            f"Expected {layout.coordinates.shape[0]} values, got {values.size}"
+        )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)